### PR TITLE
fix for #215: Fix title and nav parsing

### PR DESCRIPTION
**Pull request by AnvithLobo**
- allow macros in meta variables
- allow automatic parsing of title by mkdocs as explicitly  assigning page.title will override that.
- Fix macro parsing in nested nav `Section`s 
- Be unopinionated and let just parse macros. Let Mkdocs handle title priority

### CODE REVIEW

**Comment by fralau** [comment]
I looked at your solution and I see elegance in it (modifications are well-localized). That is definitely an advantage. 

What will matter, in final analysis, is whether it solves the issue or not, without creating new ones.

I have two objections at this stage:

1. Is that I see only processing of variables, in the rendering. What about expressions containing Jinja2 macros and filters?
2. Do you have a theory of why it would be sufficient to intervene at the stage of `on_nav`, and it is no longer necessary at `on_page_markdown`? What about the processing of page variables (e.g. defined in the page's YAML front-matter)?

File: mkdocs_macros/plugin.py [hunk 1]
```diff
@@ -571,6 +570,13 @@ def render(self, markdown: str, force_rendering:bool=False):
         # expand the template
         on_error_fail = self.config['on_error_fail']
         try:
+            # If title meta variable is present,
```
Comment (fralau) [inline mkdocs_macros/plugin.py:573]: I see only processing of variables. What about macros and filters?

Comment (AnvithLobo) [reply mkdocs_macros/plugin.py:573]: This just mimics the current behavior of the existing jinja2 rendering code for the page.
https://github.com/fralau/mkdocs-macros-plugin/blob/f97565ffe66a2b50528a00f88f36c9d390c0861b/mkdocs_macros/plugin.py#L580-L583

Unless I misread the code the `on_config` method should load all the filters and macros to `self.env` and `self.variables` right? and `page_variables` in `render` is just a copy of `self.variables`. So all macros and filters should work fine like usual. 

https://github.com/fralau/mkdocs-macros-plugin/blob/f97565ffe66a2b50528a00f88f36c9d390c0861b/mkdocs_macros/plugin.py#L696-L716

Let me know if i'm wrong. I'll look into it a bit more then.

**Comment by AnvithLobo** [comment]
> 2\. Do you have a theory of why it would be sufficient to intervene at the stage of `on_nav`, and it is no longer necessary at `on_page_markdown`? What about the processing of page variables (e.g. defined in the page's YAML front-matter)?

This is only for the nav items. Trying to render nav items from `on_page_markdown` is not ideal. As we would have to get the page and it's current `nav` then walk up the tree to find the first parent and then again walk down the tree all the way down to render the nav items. 
First this wouldn't be elegant nor efficient. As we would need to do this for every page. 
But If we only consider the last parent and try to render it's child. We will miss rendering any empty sections with no pages currently linked to that section. 

The only disadvantage of rendering Nav items on `on_nav` instead of `on_page_markdown` is that we will miss out on any meta variables defined in the page. But this is more trouble than it is worth solving IMO.

**Pull request closed without merging**